src/WizardSteps/components/WizardControls/WizardControls.tsx

```diff
@@ -68,53 +69,55 @@ const WizardControls: React.FC<WizardControlsProps> = ({
-        {controls.map((control: WizardControl, i: number) => {
-          const {
-            className,
-            label,
-            path,
-            type,
-            onClick = () => true,
-          } = control;
-          const isNext = type === 'next';
-          const isPrev = type === 'prev';
-          const controlContainerClass: string = classnames(
-            `col-sm-auto order-sm-${i}`,
-            {
-              'col-6': isNext || isPrev,
-              'col-12 mt-3 mt-sm-0 mr-auto order-last': !isNext && !isPrev,
-            }
-          );
-          const controlClass: string = classnames(
-            'btn w-100',
-            {
-              'btn-primary': isNext,
-              'btn-outline-primary': isPrev,
-              'btn-secondary': !isNext && !isPrev,
-            },
-            className
-          );
-          const controlHandler: GenericHandler = getControlHandler(type);
+        {(activeControls || controls).map(
+          (control: WizardControl, i: number) => {
+            const {
+              className,
+              label,
+              path,
+              type,
+              onClick = () => true,
+            } = control;
+            const isNext = type === 'next';
+            const isPrev = type === 'prev';
+            const controlContainerClass: string = classnames(
+              `col-sm-auto order-sm-${i}`,
+              {
+                'col-6': isNext || isPrev,
+                'col-12 mt-3 mt-sm-0 mr-auto order-last': !isNext && !isPrev,
+              }
+            );
+            const controlClass: string = classnames(
+              'btn w-100',
+              {
+                'btn-primary': isNext,
+                'btn-outline-primary': isPrev,
+                'btn-secondary': !isNext && !isPrev,
+              },
+              className
+            );
+            const controlHandler: GenericHandler = getControlHandler(type);
 
-          return (
-            <div key={label} className={controlContainerClass}>
-              <button
-                type="button"
-                className={controlClass}
-                onClick={async (event) => {
-                  /**
-                   * custom control handler, blocks navigation if result is false
-                   */
-                  if ((await onClick(event)) !== false) {
-                    controlHandler(path);
-                  }
-                }}
-              >
-                {label}
-              </button>
-            </div>
-          );
-        })}
+            return (
+              <div key={label} className={controlContainerClass}>
+                <button
+                  type="button"
+                  className={controlClass}
+                  onClick={async (event) => {
+                    /**
+                     * custom control handler, blocks navigation if result is false
+                     */
+                    if ((await onClick(event)) !== false) {
```

Comment:
can we use === true for this case? !== is slightly slower than === if not mistaken. if the result can be more than just true/false, please ignore this comment haha

Reply:
the false comparison is intention here because the `onClick` handler could be a void handler and the here conditions should only be true if `onClick` handler explicitly return a `false` value